platform/linux-generic/odp_pool.c

```diff
@@ -348,6 +348,11 @@ static odp_pool_t pool_create(const char *name, odp_pool_param_t *params,
 		break;
 
 	case ODP_POOL_PACKET:
+		if (params->pkt.min_headroom > CONFIG_PACKET_HEADROOM) {
+			ODP_ERR("Headroom size not supported");
+			return ODP_POOL_INVALID;
+		}
+
 		headroom    = CONFIG_PACKET_HEADROOM;
```

Comment:
It's not clear why we're making this change if we don't intend to do anything with it. Shouldn't this be:
`headroom = params->pkt.min_headroom;` ? 

To make this convention work correctly, `odp_pool_param_init()` needs to be updated to set `pkt.min_headroom` to `CONFIG_PACKET_HEADROOM` so that existing applications behave unchanged, which you've already done. So this looks like an omission?

Reply:
Since we are now exposing max_headroom we need to fail the pool creating if the headroom size requested could not be supported.

pkt.min_headroom is odp_pool_param_init() is set to CONFIG_PACKET_HEADROOM already. Its is the second part of this patch.

Reply:
setting headroom = CONFIG_PACKET_HEADROOM does not violate the API requirement and it also fits well with our linux-generic implementation.

Reply:
True, but I don't see why we wouldn't want to honor the application's requested pool size. Setting this unconditionally this way is just ignoring the request and if we're going to do that there's no reason to have this PR at all.

Reply:
This PR helps implementation which have the ability to maintain zero headroom size and our linux-generic implementation does not violate the API spec. We can change linux-generic in future also.

Reply:
I see no reason not to change it now, since linux-generic can support zero headroom if that's what the application requests.  Besides, this is supposed to be a reference implementation which shows how we intend APIs to behave. We intend lower headroom requests to be honored, so we should do so here.